journalbeat/checkpoint/checkpoint.go

```diff
@@ -123,6 +128,44 @@ func NewCheckpoint(file string, maxUpdates int, interval time.Duration) (*Checkp
 	return c, nil
 }
 
+// Previously the registry file was written to the root folder. It was fixed on
+// 7.x but not on 6.x. Thus, migration is needed, so users avoid losing state info.
+func (c *Checkpoint) findRegistryFile() error {
+	migratedPath := paths.Resolve(paths.Data, c.file)
+
+	fs, err := os.Stat(c.file)
+	if os.IsNotExist(err) {
+		c.file = migratedPath
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("error accessing previous registry file: %+v", err)
+	}
+
+	f, err := os.Open(c.file)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	target, err := os.OpenFile(migratedPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, fs.Mode())
+	if err != nil {
+		return err
+	}
+	defer target.Close()
+
+	if _, err := io.Copy(target, f); err != nil {
+		return err
+	}
+
+	err = helper.SafeFileRotate(c.file, c.file+".bak")
```

Comment:
We should close the files before calling `SafeFileRotate`. Also check that `Close` does not error.

Reply:
Donz.